core/src/main/java/com/sequenceiq/cloudbreak/service/stack/DefaultStackService.java

```diff
@@ -195,6 +196,28 @@ public void delete(Long id, CbUser user) {
     }
 
     @Override
+    public void removeInstance(CbUser user, Long stackId, String instanceId) {
+        Stack stack = stackRepository.findById(stackId);
+        InstanceMetaData instanceMetaData = instanceMetaDataRepository.findByInstanceId(stackId, instanceId);
+        if (stack == null) {
+            throw new NotFoundException(String.format("Not found stack with id '%s'", stackId.toString()));
+        }
```

Comment:
I think findById (and findOne) cannot return null, it throws a NotFoundException instead